RecoTauTag/RecoTau/plugins/PATTauIDEmbedder.cc

```diff
@@ -0,0 +1,95 @@
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "DataFormats/PatCandidates/interface/PATTauDiscriminator.h"
+#include "DataFormats/PatCandidates/interface/Tau.h"
+#include "FWCore/Utilities/interface/transform.h"
+
+class PATTauIDEmbedder : public edm::stream::EDProducer<>
+{
+public:
+
+	explicit PATTauIDEmbedder(const edm::ParameterSet&);
+	~PATTauIDEmbedder(){};
+
+	void produce(edm::Event&, const edm::EventSetup&);
+
+private:
+
+//--- configuration parameters
+	edm::EDGetTokenT<pat::TauCollection> src_;
+	typedef std::pair<std::string, edm::InputTag> NameTag;
+	std::vector<NameTag> tauIDSrcs_;
+	std::vector<edm::EDGetTokenT<pat::PATTauDiscriminator> > patTauIDTokens_;
+};
+
+PATTauIDEmbedder::PATTauIDEmbedder(const edm::ParameterSet& cfg)
+{
+	src_ = consumes<pat::TauCollection>(cfg.getParameter<edm::InputTag>("src"));
+	// it might be a single tau ID
+	if (cfg.existsAs<edm::InputTag>("tauIDSource")) {
+		tauIDSrcs_.push_back(NameTag("", cfg.getParameter<edm::InputTag>("tauIDSource")));
+	}
+	// or there might be many of them
+	if (cfg.existsAs<edm::ParameterSet>("tauIDSources")) {
```

Comment:
https://github.com/cms-sw/cmssw/pull/17624#discussion_r104263019 was not addressed.
if multiple input support is needed, use it by default and fill a vector of size 1 if there is just one source